### ISSUE TITLE
Installation trouble !!

### ISSUE THREAD

**Issue by akashmony88** [samuelhwilliams/Eel#214]
When I am trying to install "eel" using the "pip install eel" command I get the error...
.
.
.
The package setup script has attempted to modify files on your system
that are not within the EasyInstall build area, and has been aborted.
   
This package cannot be safely installed by EasyInstall, and may not
support alternate installation locations even if you run its setup
 script by hand.  Please inform the package's author and the EasyInstall
maintainers to find out if a fix or workaround is available.
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
.
.
.
.
How can I solve this?

**Comment by samuelhwilliams** [comment]
Hi @akashmony88 - I've just tried to reproduce this in the most basic fashion and I'm afraid I can't.

```
sam:~/git/foobar$ pip install eel
Collecting eel
  Downloading https://files.pythonhosted.org/packages/06/03/c40fc6360d2070bd680d0111bac00aa4de71a6567c758a05ac0dc4aba825/Eel-0.11.0.tar.gz
Collecting bottle (from eel)
  Downloading https://files.pythonhosted.org/packages/69/d1/efdd0a5584169cdf791d726264089ce5d96846a8978c44ac6e13ae234327/bottle-0.12.17-py3-none-any.whl (89kB)
    100% |████████████████████████████████| 92kB 583kB/s
Collecting bottle-websocket (from eel)
  Using cached https://files.pythonhosted.org/packages/17/8e/a22666b4bb0a6e31de579504077df2b1c2f1438136777c728e6cfabef295/bottle-websocket-0.2.9.tar.gz
Collecting future (from eel)
  Using cached https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz
Collecting whichcraft (from eel)
  Using cached https://files.pythonhosted.org/packages/b5/a2/81887a0dae2e4d2adc70d9a3557fdda969f863ced51cd3c47b587d25bce5/whichcraft-0.6.1-py2.py3-none-any.whl
Collecting gevent-websocket (from bottle-websocket->eel)
  Using cached https://files.pythonhosted.org/packages/7b/84/2dc373eb6493e00c884cc11e6c059ec97abae2678d42f06bf780570b0193/gevent_websocket-0.10.1-py3-none-any.whl
Collecting gevent (from gevent-websocket->bottle-websocket->eel)
  Using cached https://files.pythonhosted.org/packages/8f/e8/6bb46c451d881d02216d2219374ef69853b78d52194f622d474d49efe2ce/gevent-1.4.0-cp37-cp37m-macosx_10_9_x86_64.whl
Collecting greenlet>=0.4.14; platform_python_implementation == "CPython" (from gevent->gevent-websocket->bottle-websocket->eel)
  Using cached https://files.pythonhosted.org/packages/f8/e8/b30ae23b45f69aa3f024b46064c0ac8e5fcb4f22ace0dca8d6f9c8bbe5e7/greenlet-0.4.15.tar.gz
Installing collected packages: bottle, greenlet, gevent, gevent-websocket, bottle-websocket, future, whichcraft, eel
  Running setup.py install for greenlet ... done
  Running setup.py install for bottle-websocket ... done
  Running setup.py install for future ... done
  Running setup.py install for eel ... done
Successfully installed bottle-0.12.17 bottle-websocket-0.2.9 eel-0.11.0 future-0.18.2 gevent-1.4.0 gevent-websocket-0.10.1 greenlet-0.4.15 whichcraft-0.6.1
You are using pip version 18.1, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
sam:~/git/foobar$
```

Please can you share more details about your setup (OS, python version, eel version, exact commands, etc). Ideally please also include a full copy/paste as I have above, showing command as well as all output.

Thanks!

**Comment by akashmony88** [comment]
Hi samuelhwilliams
I am using windows 10, and the python version is 3.8. My pip is also up to
date. But still, have the problem. Here the full terminal code..
.
.
.
.
.
.
.
.
Microsoft Windows [Version 10.0.17763.864]
(c) 2018 Microsoft Corporation. All rights reserved.

(venv) D:\projects\python\aladin>pip install eel
Collecting eel
  Using cached
https://files.pythonhosted.org/packages/06/03/c40fc6360d2070bd680d0111bac00aa4de71a6567c758a05ac0dc4aba825/Eel-0.11.0.tar.gz
Collecting bottle (from eel)
  Using cached
https://files.pythonhosted.org/packages/69/d1/efdd0a5584169cdf791d726264089ce5d96846a8978c44ac6e13ae234327/bottle-0.12.17-py3-none-a
ny.whl
Collecting bottle-websocket (from eel)
  Using cached
https://files.pythonhosted.org/packages/17/8e/a22666b4bb0a6e31de579504077df2b1c2f1438136777c728e6cfabef295/bottle-websocket-0.2.9.ta
r.gz
Collecting future (from eel)
  Using cached
https://files.pythonhosted.org/packages/45/0b/38b06fd9b92dc2b68d58b75f900e97884c45bedd2ff83203d933cf5851c9/future-0.18.2.tar.gz
Collecting whichcraft (from eel)
  Using cached
https://files.pythonhosted.org/packages/b5/a2/81887a0dae2e4d2adc70d9a3557fdda969f863ced51cd3c47b587d25bce5/whichcraft-0.6.1-py2.py3-
none-any.whl
Collecting gevent-websocket (from bottle-websocket->eel)
  Using cached
https://files.pythonhosted.org/packages/7b/84/2dc373eb6493e00c884cc11e6c059ec97abae2678d42f06bf780570b0193/gevent_websocket-0.10.1-p
y3-none-any.whl
Collecting gevent (from gevent-websocket->bottle-websocket->eel)
  Using cached
https://files.pythonhosted.org/packages/ed/27/6c49b70808f569b66ec7fac2e78f076e9b204db9cf5768740cff3d5a07ae/gevent-1.4.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 154, in save_modules
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 195, in setup_context
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 250, in run_setup
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 45, in _execfile
      File
"C:\Users\TMIPR~1\AppData\Local\Temp\easy_install-y9021_ug\cffi-1.13.2\setup.py",
line 127, in <module>
        HUB_PRIMITIVES = Extension(name="gevent.__hub_primitives",
      File
"C:\Users\TMIPR~1\AppData\Local\Temp\easy_install-y9021_ug\cffi-1.13.2\setup.py",
line 105, in uses_msvc
        include_dirs=include_dirs)
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\distutils\command\config.py",
line 225, in try_compile
        self._compile(body, headers, include_dirs, lang)
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\distutils\command\config.py",
line 127, in _compile
        src = self._gen_temp_sourcefile(body, headers, lang)
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\distutils\command\config.py",
line 109, in _gen_temp_sourcefile
        with open(filename, "w") as file:
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 418, in _open
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 407, in _violation
    setuptools.sandbox.SandboxViolation: SandboxViolation:
open('_configtest.c', 'w') {}

    The package setup script has attempted to modify files on your system
    that are not within the EasyInstall build area, and has been aborted.

    This package cannot be safely installed by EasyInstall, and may not
    support alternate installation locations even if you run its setup
    script by hand.  Please inform the package's author and the EasyInstall
    maintainers to find out if a fix or workaround is available.


    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\tmi
pr\AppData\Local\Temp\pip-install-wyrro1i5\gevent\setup.py", line 427, in
<module>
        run_setup(EXT_MODULES, run_make=_BUILDING)
      File "C:\Users\tmi
pr\AppData\Local\Temp\pip-install-wyrro1i5\gevent\setup.py", line 328, in
run_setup
        setup(
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\__init__.py",
line 144, in setup
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\__init__.py",
line 139, in _install_setup_requi
res
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\dist.py",
line 721, in fetch_build_eggs
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\pkg_resources\__init__.py",
line 780, in resolve
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\pkg_resources\__init__.py",
line 1065, in best_match
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\pkg_resources\__init__.py",
line 1077, in obtain
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\dist.py",
line 791, in fetch_build_egg
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\command\easy_install.py",
line 679, in easy_ins
tall
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\command\easy_install.py",
line 705, in install_
item
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\command\easy_install.py",
line 890, in install_
eggs
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\command\easy_install.py",
line 1158, in build_a
nd_install
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\command\easy_install.py",
line 1144, in run_set
up
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 253, in run_setup
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\contextlib.py", line 131, in
__exit__
        self.gen.throw(type, value, traceback)
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 195, in setup_context
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\contextlib.py", line 131, in
__exit__
        self.gen.throw(type, value, traceback)
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 166, in save_modules
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 141, in resume
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\_vendor\six.py",
line 685, in reraise
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 154, in save_modules
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 195, in setup_context
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 250, in run_setup
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 45, in _execfile
      File
"C:\Users\TMIPR~1\AppData\Local\Temp\easy_install-y9021_ug\cffi-1.13.2\setup.py",
line 127, in <module>
        HUB_PRIMITIVES = Extension(name="gevent.__hub_primitives",
      File
"C:\Users\TMIPR~1\AppData\Local\Temp\easy_install-y9021_ug\cffi-1.13.2\setup.py",
line 105, in uses_msvc
        include_dirs=include_dirs)
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\distutils\command\config.py",
line 225, in try_compile
        self._compile(body, headers, include_dirs, lang)
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\distutils\command\config.py",
line 127, in _compile
        src = self._gen_temp_sourcefile(body, headers, lang)
      File "C:\Users\tmi
pr\AppData\Local\Programs\Python\Python38\lib\distutils\command\config.py",
line 109, in _gen_temp_sourcefile
        with open(filename, "w") as file:
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 418, in _open
      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 407, in _violation
    setuptools.sandbox.SandboxViolation: SandboxViolation:
open('_configtest.c', 'w') {}

    The package setup script has attempted to modify files on your system
    that are not within the EasyInstall build area, and has been aborted.

    This package cannot be safely installed by EasyInstall, and may not
    support alternate installation locations even if you run its setup
    script by hand.  Please inform the package's author and the EasyInstall
    maintainers to find out if a fix or workaround is available.


    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in C:\Users\tmi
pr\AppData\Local\Temp\pip-install-wyrro1i5\gevent\

(venv)
D:\projects\python\aladin>cts\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
line 418, in _open
The system cannot find the path specified.

(venv) D:\projects\python\aladin>      File
"D:\projects\python\aladin\venv\lib\site-packages\setuptools-40.8.0-py3.8.egg\setuptools\sandbox.py",
l
ine 407, in _violation
'File' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>    setuptools.sandbox.SandboxViolation:
SandboxViolation: open('_configtest.c', 'w') {}
'setuptools.sandbox.SandboxViolation:' is not recognized as an internal or
external command,
operable program or batch file.

(venv) D:\projects\python\aladin>
(venv) D:\projects\python\aladin>    The package setup script has attempted
to modify files on your system
'The' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>    that are not within the EasyInstall
build area, and has been aborted.
'that' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>
(venv) D:\projects\python\aladin>    This package cannot be safely
installed by EasyInstall, and may not
'This' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>    support alternate installation
locations even if you run its setup
'support' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>    script by hand.  Please inform the
package's author and the EasyInstall
'script' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>    maintainers to find out if a fix or
workaround is available.
'maintainers' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>
(venv) D:\projects\python\aladin>
(venv) D:\projects\python\aladin>
 ----------------------------------------
'----------------------------------------' is not recognized as an internal
or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>Command "python setup.py egg_info" failed
with error code 1 in C:\Users\tmi pr\AppData\Local\Temp\pip-install-wyrr
o1i5\gevent\
'Command' is not recognized as an internal or external command,
operable program or batch file.

(venv) D:\projects\python\aladin>
(venv) D:\projects\python\aladin>(venv) D:\projects\python\aladin>
D:\projects\python\aladin was unexpected at this time.
(venv) D:\projects\python\aladin>
.
.
.
.
.
.
Thanks From
*Mohammed Akash*

Phone: 0193135824
Email: akashmonya88@gmail.com

Facebook: https://www.facebook.com/*akashmony88*
<https://www.facebook.com/akashmony88>
Twitter: https://twitter.com/*akashmony88* <https://twitter.com/akashmony88>
Linked In: https://www.linkedin.com/in/*akashmony88*
<https://www.linkedin.com/in/akashmony88>
Github: https://github.com/*akashmony88* <https://github.com/akashmony88>

**Comment by samuelhwilliams** [comment]
I think this is not strictly an issue with Eel, but a recent regression affecting Python 3.8 on Windows setuptools < 42.0.0.

This is based on a little googling, which showed up the following:
* https://github.com/numpy/numpy/issues/14873
* https://github.com/pypa/setuptools/issues/1874
* https://setuptools.readthedocs.io/en/latest/history.html#v42-0-0

I think it was fixed in setuptools 42.0.0, so possibly upgrading to that will help. An alternative workaround may be to downgrade to Python 3.7.x. I'm not sure I see that there's anything to be done in Eel at the moment though.

**Comment by samuelhwilliams** [comment]
I'm going to close this for now based on my above comments, however if you continue to have problems please ping me here and we can look into it again/further.

**Comment by jrcribb** [comment]
Check the error list searching a line like 'WARNING: The wheel package is not available.' or something similar.
Installing separately each missing package resolve the problems.

**Comment by the91end** [comment]
Thanks
I fixed the issue by upgrading setuptools

> I think this is not strictly an issue with Eel, but a recent regression affecting Python 3.8 on Windows setuptools < 42.0.0.
> 
> This is based on a little googling, which showed up the following:
> 
> * [numpy/numpy#14873](https://github.com/numpy/numpy/issues/14873)
> * [pypa/setuptools#1874](https://github.com/pypa/setuptools/issues/1874)
> * https://setuptools.readthedocs.io/en/latest/history.html#v42-0-0
> 
> I think it was fixed in setuptools 42.0.0, so possibly upgrading to that will help. An alternative workaround may be to downgrade to Python 3.7.x. I'm not sure I see that there's anything to be done in Eel at the moment though.
